docker context is not respected

### What happened?

Running `gefyra status`:
```
[CRITICAL] Docker init error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
[ERROR] Could not create a valid configuration: Docker init error. Docker host not running?
[INFO] There was no --endpoint argument provided. Connecting to a local Kubernetes node.
[CRITICAL] Docker init error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
[ERROR] Could not create a valid configuration: Docker init error. Docker host not running?
[CRITICAL] Docker init error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
[CRITICAL] There was an error running Gefyra: Docker init error. Docker host not running?
```

### What did you expect to happen?

I am using a non-default docker context. 
Running `docker context ls`
```
NAME              DESCRIPTION                               DOCKER ENDPOINT                                         KUBERNETES ENDPOINT                ORCHESTRATOR
colima *          colima                                    unix:///Users/ventsislavg/.colima/default/docker.sock
default           Current DOCKER_HOST based configuration   unix:///var/run/docker.sock                             https://127.0.0.1:6443 (default)   swarm
```

### How can we reproduce it (as minimally and precisely as possible)?

on macOS:
```
brew install colima
colima start
docker info
gefyra status
```

### What Kubernetes setup are you working with?

N/A

### OS version

Darwin ventsislavg 22.1.0 Darwin Kernel Version 22.1.0: Tue Sep 27 22:08:45 PDT 2022; root:xnu-8792.41.6~5/RELEASE_ARM64_T6000 arm64

### Anything else we need to know?

_No response_

--- a/client/gefyra/configuration.py
+++ b/client/gefyra/configuration.py
@@ -161,9 +161,16 @@ class ClientConfiguration(object):
 
     def _init_docker(self):
         import docker
+        from docker.context import ContextAPI
 
         try:
-            self.DOCKER = docker.from_env()
+            ctx = ContextAPI.get_context()
+            if ctx.name != "default":
+                endpoint = ctx.endpoints["docker"]["Host"]
+                self.DOCKER = docker.DockerClient(base_url=endpoint)
+                logger.debug(f"Docker Context: {ctx.name}")
+            else:
+                self.DOCKER = docker.from_env()
         except docker.errors.DockerException as de:
             logger.fatal(f"Docker init error: {de}")
             raise docker.errors.DockerException(

